library/entropy_poll.c

```diff
@@ -175,19 +198,33 @@ int mbedtls_platform_entropy_poll( void *data,
     }
     else if( errno != ENOSYS )
         return( MBEDTLS_ERR_ENTROPY_SOURCE_FAILED );
-    /* Fall through if the system call isn't known. */
-#else
-    ((void) ret);
+    else
```

Comment:
I don't understand why this fall-through has changed? Returning ret (which at this point will be -1) is wrong, as `mbedtls_platform_entropy_poll` should return 0 on success, or one of the MBEDTLS_ERR_ENTROPY_xxx error codes (according to the prototype in entropy.h:101).